notebooks/create_synthetic_data.py

```
@@ -11,10 +11,10 @@
 spark = SparkSession.builder.getOrCreate()
 
 try:
-    table_path = f"{config.catalog}.{config.db_schema}.{config.use_case_name}"
-    full_data = spark.read.table(table_path)
+    input_table_path = f"{config.catalog}.{config.db_schema}.{config.use_case_name}"
+    full_data = spark.read.table(input_table_path)
 except Exception as e:
-    raise RuntimeError(f"Failed to read table {table_path}: {str(e)}")
+    raise RuntimeError(f"Failed to read table {input_table_path}: {str(e)}")
```

Ruff (0.7.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)